mat/io.go

```diff
@@ -370,3 +427,53 @@ func readFull(r io.Reader, buf []byte) (int, error) {
 	}
 	return n, err
 }
+
+// storage is the internal representation of the storage format of a
+// serialised matrix.
+type storage struct {
+	Version uint32 // Keep this first.
+	Form    byte   // [GST]
+	Packing byte   // [BPF]
+	Uplo    byte   // [AUL]
+	Unit    bool
+	Rows    int64
+	Cols    int64
+	KU      int64
+	KL      int64
+}
+
+func (s storage) marshalBinaryTo(w io.Writer) (int, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, headerSize))
+	err := binary.Write(buf, binary.LittleEndian, s)
```

Comment:
this is using a lot of (slow) reflection.
could we at least add a `TODO/FIXME` to replace this with direct calls to `LittleEndian.PutUintXX` ?

Reply:
Yeah, this is a first iteration, so since I didn't have any indication of whether the proposed approach was OK, I did the simplest (for me) to write. These can be optimised, though if you I/O is your major issue, you're lucky.

Reply:
Added TODO.